### PR TITLE
feat: open gated model access pages in the Comfy session

### PR DESCRIPTION
**Pull request by jaeone94**
## Summary

This is the Desktop half of the gated Hugging Face model access flow introduced in [Comfy-Org/ComfyUI_frontend#13742](https://github.com/Comfy-Org/ComfyUI_frontend/pull/13742).

The frontend PR adds a lock action beside missing models that appear to require Hugging Face authorization while leaving the normal Download action intact. This paired PR adds an explicit `openModelAccessPage()` Desktop bridge implementation so that action opens the repository in a Hugging Face-only Desktop window using the same Electron session as the originating ComfyUI view.

Sharing the session is the important part: after the user signs in and accepts the repository terms, the existing Desktop model download manager uses that same session for `session.downloadURL()`. The user can return to ComfyUI, click Download again, and complete the download without a separate OAuth or token flow.

## Changes

- Expose `openModelAccessPage(url)` from the ComfyUI preload bridge.
- Register a dedicated `desktop2-open-model-access-page` IPC handler.
- Resolve the host window and Electron session from the IPC sender rather than using a global browser session.
- Open the repository in a child `BrowserWindow` configured with the sender's session, no preload, `nodeIntegration: false`, `contextIsolation: true`, and `sandbox: true`.
- Accept only exact `https://huggingface.co/<owner>/<repository>` model repository URLs on the default HTTPS port as bridge entry points.
- Allow navigation anywhere within the exact `https://huggingface.co` origin after the trusted repository window opens. This keeps Hugging Face home, search, settings, sign-in, sign-out, license, Spaces, and other repository flows usable.
- Block HTTP, lookalike hosts, subdomains, cross-origin navigation, new Electron popups, and browser-permission requests from the remote access page.
- Reuse an existing window only for the same Comfy sender and repository. Separate repositories and separate sender sessions cannot overwrite or share an access flow.
- Destroy access windows with their owning Comfy host and remove lifecycle listeners when a child closes.
- Capture the access page `webContents` while the window is alive so the `closed` handler can finish cleanup without dereferencing a destroyed `BrowserWindow`.
- Treat a superseded `ERR_ABORTED` navigation as accepted, while returning `false` for genuine initial load failures so the frontend can run its browser fallback.
- Add focused tests for the preload/IPC contract, URL and origin boundaries, session ownership, lifecycle cleanup, load failures, popup denial, and permission handling.

## Paired Frontend Contract

The frontend contract is defined by [Comfy-Org/ComfyUI_frontend#13742](https://github.com/Comfy-Org/ComfyUI_frontend/pull/13742) as an optional bridge method:

```ts
openModelAccessPage?: (url: string) => Promise<boolean>
```

The method resolves `true` when Desktop takes ownership of the trusted access-page request. It resolves `false` when the URL is not trusted, the originating ComfyUI host is unavailable, or the initial page load genuinely fails. A `false` result allows the frontend to fall back to opening the page in the normal browser.

Desktop currently consumes the published bridge-types package at `0.1.2`, while the frontend PR advances that package to `0.1.4`. The preload uses a small local intersection type so both PRs can be reviewed and tested in parallel. Once `0.1.4` is published, Desktop can bump the package and remove that temporary local type without changing runtime behavior.

## User Flow

1. The user clicks Download for a gated model and the Desktop download fails because the session does not yet have access.
2. The user clicks the lock icon added by the paired frontend PR.
3. Desktop opens the actual Hugging Face repository in a window backed by the same Electron session as ComfyUI.
4. The user signs in with Hugging Face and accepts the repository's access terms.
5. The user returns to ComfyUI and clicks Download again.
6. The existing Desktop download manager receives the authorized session cookies and downloads the model normally.

## Intended Scope and Limitations

- Desktop does not independently classify a model as gated; it consumes the repository URL hint produced by the frontend.
- The bridge remains a narrow model-repository entry point, not an arbitrary URL opener. Once opened, the hardened remote window acts as a constrained first-party Hugging Face browser and permits navigation only within the exact Hugging Face origin.
- Third-party identity-provider redirects and popup-based enterprise SSO are not enabled. Cross-origin navigation and all renderer-created windows remain blocked.
- This does not add Hugging Face OAuth, token storage, provider API calls, model discovery, or new download-manager states.
- The access window does not automatically retry a failed model download. The user returns to ComfyUI and clicks Download again.
- The feature depends on the paired frontend UI; older frontends simply do not call the optional bridge method.

## Review Focus

- Confirm that both the access window and `startModelDownload()` use the originating ComfyUI sender session.
- Confirm that bridge entry validation happens before `BrowserWindow` creation and cannot be bypassed with HTTP, a lookalike hostname, a subdomain, a reserved route, or a non-repository path.
- Confirm that post-entry navigation is allowed throughout the exact `https://huggingface.co` origin while all other origins remain blocked.
- Confirm that the remote page receives no Desktop preload, Node.js capability, popup surface, or browser permissions.
- Confirm that access windows cannot outlive their host or cross sender-session boundaries.
- Confirm that genuine load failures return `false` while superseded navigation remains open.
- Confirm that this remains a provider-access bridge rather than a second model download implementation.

## Validation

- `pnpm run typecheck`
- `pnpm run lint`
- `pnpm run build`
- `pnpm run test` (`190` files, `2,778` passed, `1` skipped)
- `pnpm run test:integration` (`25` passed)
- `pnpm exec vitest run src/main/lib/modelAccessPage.test.ts` (`25` passed, including destroyed-window cleanup)
- Manually verified the normal Hugging Face access-grant and retry flow using the paired frontend production build in Comfy Desktop.
- Manually verified Hugging Face sign-out and same-origin navigation in the embedded access window after replacing the path allowlist with an exact-origin boundary.
- Reproduced an immediate login navigation as `ERR_ABORTED (-3)` in Electron and verified that the access window reaches the login page while the main process remains responsive.

An automated E2E test is not included because this flow requires a real third-party Hugging Face account, a gated repository access grant, and persistent cookies in a running Electron session. Unit tests cover the deterministic bridge, trust boundary, lifecycle, and session wiring; the authenticated provider flow was verified manually.

## Screenshots

https://github.com/user-attachments/assets/56d015f3-fb2d-443d-96ae-6b7617f130d6

### CODE REVIEW

File: src/main/lib/modelAccessPage.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    return parsed
+  } catch {
+    return null
+  }
+}
+
+function modelRepositoryPath(url: URL): string | null {
+  const [owner, repository] = url.pathname.split('/').filter(Boolean)
+  if (!owner || !repository || owner === 'spaces') return null
+  return `/${owner}/${repository}`
```
[the remark this answers is not in the document]
Comment (jaeone94) [reply src/main/lib/modelAccessPage.ts:53]: Addressed in 9eb0301. Initial bridge requests now require an exact two-segment model repository path and reject Hugging Face reserved namespaces, with table coverage for datasets, login, settings, Spaces, and nested file routes.

**Review by benceruleanlu — DISMISSED** [review]

**Comment by benceruleanlu** [comment]
Hold on a second actually

**Review by benceruleanlu — COMMENTED** [review]
Below verified manually

**Bug: closing the Hugging Face window throws `TypeError: Object has been destroyed` in the main process**

The `closed` handler's first line accesses `accessWindow.webContents`, but the window is already destroyed when `closed` fires — verified on Electron 40.4.1 for both `close()` and `destroy()`. Every close path hits it, including the happy path (sign in, accept terms, close the window).

The app only registers `uncaughtExceptionMonitor`, so the exception goes unhandled and packaged builds show Electron's native "A JavaScript error occurred in the main process" dialog on every close. The throw also skips the rest of the handler (stale `accessWindows` entry, leaked parent `closed` listener). Tests miss it because the mocked `webContents` never throws after destroy.

Fix — capture the reference while the window is alive:

```ts
const accessContents = accessWindow.webContents
accessWindow.once('closed', () => {
  accessPageContents.delete(accessContents)
  ...
})
```

**Comment by jaeone94** [comment]
@benceruleanlu Thanks for catching and manually verifying this. Fixed in `55c1cc8`: the access-page `webContents` reference is now captured while the window is alive, so the `closed` handler no longer dereferences a destroyed `BrowserWindow` and the remaining map/listener cleanup can complete. I also added a regression test that makes post-destruction `webContents` access throw `Object has been destroyed` and verifies the close handler still finishes. The focused suite now passes 25/25, and the full typecheck and lint gates pass.

File: src/main/lib/modelAccessPage.test.ts [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+  it('cleans up after Electron destroys the access window web contents', async () => {
+    await openModelAccessPageWindow({ session } as unknown as WebContents, url)
+    Object.defineProperty(accessWindow, 'webContents', {
+      configurable: true,
+      get: () => {
+        throw new TypeError('Object has been destroyed')
+      }
+    })
+
+    expect(() => accessWindowListeners.get('closed')?.()).not.toThrow()
```
[the remark this answers is not in the document]
Comment (jaeone94) [reply src/main/lib/modelAccessPage.test.ts:179]: Thanks. The following assertion verifies that `parent.removeListener` was called from inside the `closed` handler, so the test still fails if the listener is missing or not invoked. I’m keeping the optional chaining consistent with the other mocked event triggers in this test file.

**Review by benceruleanlu — APPROVED** [review]
Nothing blocking
